### ISSUE TITLE
Path selection not working with 31 folders

### ISSUE THREAD

**Issue by phuongnq** [craftercms/craftercms#6988]
### CI Number

5150

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] The issue is in the develop branch
- [X] The issue is in the latest released 4.1.x
- [ ] The issue is in the latest released 4.0.x
- [ ] The issue is in the latest released 3.1.x

### Describe the issue

We currently have an issue with the Path selection of pages when there are 31 subfolders. The last folder is not shown.

### Steps to reproduce

Steps:

1. Create a new site from Empty blueprint
2. Using bash and git client to add 31 folders to the `/site/website/` folder and commit. This can be done via a simple bash script such as:
```
for i in {1..31}
do
  mkdir "folder$i"
  touch "folder$i/.keep"
done
```
3. From the UI, select Search screen > Path > Select Path  and browse to `/site/website` (Home item)
4. Notice that there are 31 folders but only 30 folders are displayed and there is no '...1 more item' link. This is not an expected behavior.

In this screenshot, `folder9` is missing:
![2024-06-21_10-00](https://github.com/craftercms/craftercms/assets/2996543/05b4b516-6117-492d-a0b0-e2e5e1f46f51)

While browsing the sidebar, the `folder9` does exist:
![2024-06-21_10-00_1](https://github.com/craftercms/craftercms/assets/2996543/bc16e841-c5c3-4fa7-a139-0173663b49c5)

6. Add one more folder such as `folder32` to the path `/site/website` and repeat steps 3, 4. 
7. Notice that there is '...1 more item' link but actually, we have 2 more items. This is also unexpected behavior.

In this screenshot, `folder8` and `folder9` do not show up and the link only says 1 more item but it has 2 more items. If we click the link, it shows 2 items.
![2024-06-21_10-01](https://github.com/craftercms/craftercms/assets/2996543/510416e5-3d7c-4888-aebb-e271af4a7f04)

**Comment by jvega190** [comment]
This ticket depends on https://github.com/craftercms/craftercms/issues/6992, as the response needs to be fixed, and a UI update may be needed here

**Comment by Euquimides** [comment]
Hi! Fix verified for v4.2. Thank you!

**Comment by yacdaniel** [comment]
Verified fixed in `4.1.6-SNAPSHOT`
